File: examples/chicken_wrap.pl
#!/usr/bin/env perl

# To setup this script, you need a modern Perl installation with the FFI::Platypus module installed.
# Create the libchickenize.so dynamic library by running:
#   cargo build --release
# And then execute this Perl example from the top directory as:
#   perl examples/chicken_wrap.pl

use v5.10;
use strict;
use warnings;
use FFI::Platypus;

my $ffi = FFI::Platypus->new;
$ffi->lib("target/release/libchickenize.so");
# attach as a xsub and call (much faster)
$ffi->attach( chickenize => ['string'] => 'string' );
$ffi->attach( buffalo => ['string'] => 'string' );
$ffi->attach( lorem_ipsum => ['string'] => 'string' );
$ffi->attach( anonymize => ['string','string'] => 'string' );

my $example = "The Vice-president didn't like his style";

say "";
say "Chickenized:";
say chickenize($example);

say "";
say "Buffalo:";
say buffalo($example);

say "";
say "Lorem:";
say lorem_ipsum($example);

say "";
say "Anonymized (meow):";
say anonymize($example, "meow");
say "";

File: examples/chicken_wrap.rs
// If you want to try something cool, follow the instructions to install asm.js here:
// https://users.rust-lang.org/t/compiling-to-the-web-with-rust-and-emscripten/7627
//
// and natively compile to JS via cargo:
//
// cargo build --release --target=asmjs-unknown-emscripten --example chicken_wrap
//
// Try it out via:
// node target/asmjs-unknown-emscripten/release/examples/chicken_wrap.js
//
extern crate chickenize;
use chickenize::*;

fn main() {
  let example = "The Vice-president didn't like his style";

  println!("");
  println!("Chickenized:");
  println!("{:?}",example.chicken());

  println!("");
  println!("Buffalo:");
  println!("{:?}",example.buffalo());

  println!("");
  println!("Lorem:");
  println!("{:?}",example.lorem_ipsum());

  println!("");
  println!("Anonymized (meow):");
  println!("{:?}",example.anonymize("meow"));
  println!("");
}